src/main/java/teammates/logic/automated/FeedbackSessionUnpublishedMailAction.java

```diff
@@ -0,0 +1,85 @@
+package teammates.logic.automated;
+
+import java.util.HashMap;
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+
+import teammates.common.datatransfer.FeedbackSessionAttributes;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.common.util.Assumption;
+import teammates.common.util.Const;
+import teammates.common.util.Const.ParamsNames;
+import teammates.common.util.EmailWrapper;
+import teammates.common.util.HttpRequestHelper;
+import teammates.logic.core.EmailGenerator;
+import teammates.logic.core.FeedbackSessionsLogic;
+
+public class FeedbackSessionUnpublishedMailAction extends EmailAction {
+
+    private String feedbackSessionName;
+    private String courseId;
+    
+    public FeedbackSessionUnpublishedMailAction(HttpServletRequest request) {
+        super(request);
+        initializeNameAndDescription();
+        
+        feedbackSessionName = HttpRequestHelper
+                .getValueFromRequestParameterMap(request, ParamsNames.EMAIL_FEEDBACK);
+        Assumption.assertNotNull(feedbackSessionName);
+        
+        courseId = HttpRequestHelper
+                .getValueFromRequestParameterMap(request, ParamsNames.EMAIL_COURSE);
+        Assumption.assertNotNull(courseId);
+    }
+    
+    public FeedbackSessionUnpublishedMailAction(HashMap<String, String> paramMap) {
+        super();
+        initializeNameAndDescription();
+        
+        feedbackSessionName = paramMap.get(ParamsNames.EMAIL_FEEDBACK);
+        Assumption.assertNotNull(feedbackSessionName);
+        
+        courseId = paramMap.get(ParamsNames.EMAIL_COURSE);
+        Assumption.assertNotNull(courseId);
+    }
+    
+    @Override
+    protected void doPostProcessingForSuccesfulSend() throws InvalidParametersException, EntityDoesNotExistException {
+        FeedbackSessionAttributes feedbackSession = FeedbackSessionsLogic.inst()
+                .getFeedbackSession(feedbackSessionName, courseId);
+        feedbackSession.setSentPublishedEmail(false);
+        FeedbackSessionsLogic.inst().updateFeedbackSession(feedbackSession);
+    }
+
+    @Override
+    protected List<EmailWrapper> prepareMailToBeSent() {
+        
+        FeedbackSessionAttributes feedbackSession = FeedbackSessionsLogic.inst()
+                .getFeedbackSession(feedbackSessionName, courseId);
+        log.info("Fetching feedback session object for feedback session name : "
```

Comment:
A minor point, but I think we can omit this. This may have made sense in the past before we had email logs, but it is likely not helpful anymore.